### PR TITLE
chore: upgrade `eslint` to `^8.0.0`

### PR DESCRIPTION
**Pull request by faustbrian**
Upgrades `eslint` to the latest major release

### CODE REVIEW

**Comment by holgerd77** [comment]
The man who does all the work at once. 😂

(but great thing to tackle, maybe I've got too much respect for these kind of work tasks... 😜)

File: packages/client/.eslintrc.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   extends: '../../config/eslint.js',
   parserOptions: {
+    extraFileExtensions: ['.json'],
```
Comment (holgerd77) [inline packages/client/.eslintrc.js:4]: Does this include/activate linting on JSON files?

Comment (holgerd77) [reply packages/client/.eslintrc.js:4]: Ah no, I think linting for JSON is great. 👍 (not sure if there are other opinions here)

**Review by acolytec3 — CHANGES_REQUESTED** [review]
Looks good.  One deletion of archaic references and this looks ready for merging to me.

**Comment by acolytec3** [comment]
> @acolytec3 what should be deleted?

Ugh, sorry, VS Code ate my comments again. The `env: mocha` in the eslint config in `rlp`

**Review by acolytec3 — APPROVED** [review]
LGTM.  @holgerd77 anything else to add?

**Comment by holgerd77** [comment]
> @holgerd77 [88c2563](https://github.com/ethereumjs/ethereumjs-monorepo/commit/88c25636d322716a962136885b461b617bea9816)

Also the name in the header title, so `# rlp` -> `# @ethereumjs/rlp`

**Comment by holgerd77** [comment]
@faustbrian we are doing our team call today at 5:30pm Berlin time, if you are interested you are very invited to join, then I'll send you a link.

File: packages/client/lib/rpc/modules/engine.ts [hunk 2]
```diff
@@ -511,7 +511,7 @@ export class Engine {
             headBlock.hash()
           )}`
         )
-        this.service.beaconSync?.setHead(headBlock)
+        await this.service.beaconSync?.setHead(headBlock)
```
Comment (holgerd77) [inline packages/client/lib/rpc/modules/engine.ts:514]: Oh, linting already at our service (or is this no rule and you side-discovered?) 🙂

Comment (holgerd77) [reply packages/client/lib/rpc/modules/engine.ts:514]: But it wasn't added in this PR or was it (can't find)? (just out of interest, wonder how this was triggered?)

Comment (holgerd77) [reply packages/client/lib/rpc/modules/engine.ts:514]: Ah, then it's likely some bug in their parser on how to find this locations I guess.

**Review by holgerd77 — APPROVED** [review]
LGTM

File: packages/client/lib/rpc/modules/engine.ts (hunk 2 above)
Comment (holgerd77) [reply packages/client/lib/rpc/modules/engine.ts:514]: Thanks a lot for the upgrade here, just merged! 🙂

**Pull request merged**